graphql-server/Sources/App/Application+configure.swift

```diff
@@ -0,0 +1,30 @@
+import Foundation
+import Hummingbird
+import HummingbirdFoundation
+import JSONValueRX
+import GraphQL
+
+extension HBApplication {
+    public func configure() throws {
+        encoder = JSONEncoder()
+        decoder = JSONDecoder()
+        
+        // MARK: - GraphQL Hummingbird Extension
+        let graphQLHandler = try GraphQLHandler(group: self.eventLoopGroup)
+        extensions.set(\.graphQLHandler, value: graphQLHandler)
+        
+        // MARK: - Routes
+        router.post("/graphql", body: .collate) { request -> EventLoopFuture<GraphQLResult> in
+            guard let json = try? request.decode(as: JSONValue.self),
```

Comment:
Is it necessary to bring in the `JSONValue` package here. It doesn't seem to give us a great deal over
```swift
struct Query {
    let query: String
}
guard let query = try? request.decode(as: Query.self) else { 
    // also note the shortcut to request.eventLoop.makeSucceededFuture.
    return request.success(GraphQLResult.invalidRequest) 
}
return graphQLHandler.handle(query: query.query)
```

Reply:
Even better: I’m using GraphQL’s internal `Map` type which feels pretty good. Thoughts?

Reply:
Looks good